src/cpp/fastdds/xtypes/dynamic_types/DynamicDataImpl.cpp

```diff
@@ -0,0 +1,7218 @@
+                        value_.erase(it);
+                        value_.emplace(id, value->clone());
+                        return RETCODE_OK;
+                    }
+                }
+
+                if (RETCODE_OK == ret_value && TK_UNION == type_kind && 0 != id) // Set new discriminator.
+                {
+                    set_discriminator_value(id);
+                }
+            }
+        }
+        else if (TK_ARRAY == type_kind ||
+                TK_SEQUENCE == type_kind)
+        {
+            TypeKind element_kind =
+                    get_enclosing_typekind(traits<DynamicType>::narrow<DynamicTypeImpl>(
+                                enclosing_type_->get_descriptor().element_type()));
+            if (is_complex_kind(element_kind))
+            {
+                auto it = value_.cbegin();
+                assert(value_.cend() != it && MEMBER_ID_INVALID == it->first);
+                auto sequence =
+                        std::static_pointer_cast<std::vector<traits<DynamicData>::ref_type>>(it->second);
+                assert(sequence);
+                if ((TK_ARRAY == type_kind && sequence->size() >= id + 1) ||
+                        (TK_SEQUENCE == type_kind &&
+                        (static_cast<uint32_t>(LENGTH_UNLIMITED) == enclosing_type_->get_descriptor().bound().at(0) ||
+                        enclosing_type_->get_descriptor().bound().at(0) >= id + 1)))
+                {
+                    if (sequence->size() < id + 1)
+                    {
+                        sequence->resize(id + 1);
+                    }
+
+                    auto pos = sequence->begin() + id;
+                    auto data = std::static_pointer_cast<DynamicDataImpl>(*pos);
+                    if (data->type_->equals(value->type()))
+                    {
+                        pos = sequence->erase(pos);
+                        sequence->emplace(pos, value->clone());
+                        ret_value =  RETCODE_OK;
+                    }
+                }
+            }
+        }
+        else if (TK_MAP == type_kind)
+        {
+            TypeKind element_kind =
+                    get_enclosing_typekind(traits<DynamicType>::narrow<DynamicTypeImpl>(
+                                enclosing_type_->get_descriptor().element_type()));
+            if (is_complex_kind(element_kind))
+            {
+                auto it = value_.find(id);
+                if (it != value_.end())
+                {
+                    auto data = std::static_pointer_cast<DynamicDataImpl>(it->second);
+
+                    if (data->type_->equals(value->type()))
+                    {
+                        value_.erase(it);
+                        value_.emplace(id, value->clone());
+                        ret_value =  RETCODE_OK;
+                    }
+                }
+            }
+        }
+    }
+    else
+    {
+        EPROSIMA_LOG_ERROR(DYN_TYPES, "Error getting complex value. Invalid MemberId.");
```

Comment:
This is not the only case where this log is displayed. Also if the value reference passed by the user is uninitialized.

```suggestion
        EPROSIMA_LOG_ERROR(DYN_TYPES, "Error setting complex value. Invalid MemberId.");
```

Reply:
*std::static_pointer_cast<TypeForKind<TK_STRING8>>(value_iterator->second) = value